app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author_id
+  belongs_to :author_id, class_name: 'User'
+  has_many :likes, foreign_key: 'post_id'
+  has_many :comments, foreign_key: 'post_id'
+
+  def update_post_counter
+    user = User.find(author_id.id)
+    user.post_counter ? user.post_counter += 1 : user.post_counter = 1
+    user.save
+  end
```

Comment:
- [ ] [OPTIONAL] Please consider using `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.

- [ ] [OPTIONAL] You can use the following method body instead of the one you have which has the same functionality.
``` 
user.increment!(:post_counter)
```
(You can consider doing this to all other similar methods in other controllers

Reply:
@ShadyShawkat  Thanks a lot for your feedback, I implement what you just mentioned and now the code look way cleaner thanks again for your sugession.